test/helpers/block.ts

```diff
@@ -0,0 +1,378 @@
+import "@moonbeam-network/api-augment/moonbase";
+import type { RuntimeDispatchInfoV1 } from "@polkadot/types/interfaces/payment";
+import { ApiPromise } from "@polkadot/api";
+import { BlockHash, DispatchInfo, RuntimeDispatchInfo } from "@polkadot/types/interfaces";
+import { u32, u128, Option } from "@polkadot/types";
+import {
+  WEIGHT_PER_GAS,
+  mapExtrinsics,
+  BlockRangeOption,
+  calculateFeePortions,
+  EXTRINSIC_BASE_WEIGHT,
+} from "@moonwall/util";
+import { DevModeContext } from "@moonwall/cli";
+import { expect } from "@moonwall/cli";
+import type { Block, AccountId20 } from "@polkadot/types/interfaces/runtime/types";
+import type { TxWithEvent } from "@polkadot/api-derive/types";
+import type { ITuple } from "@polkadot/types-codec/types";
+
+const debug = require("debug")("test:blocks");
+
+// Given a deposit amount, returns the amount burned (80%) and deposited to treasury (20%).
+// This is meant to precisely mimic the logic in the Moonbeam runtimes where the burn amount
+// is calculated and the treasury is treated as the remainder. This precision is important to
+// avoid off-by-one errors.
+
+export interface TxWithEventAndFee extends TxWithEvent {
+  fee: RuntimeDispatchInfo | RuntimeDispatchInfoV1;
+}
+
+export interface BlockDetails {
+  block: Block;
+  txWithEvents: TxWithEventAndFee[];
+}
+
+const getBlockDetails = async (
```

Comment:
I think the functions in this file should definitely go to moonwall somehow and be more generic.